src/eureka/S3_data_reduction/s3_reduce.py

```diff
@@ -652,6 +652,11 @@ def reduce(eventlabel, ecf_path=None, s2_meta=None, input_meta=None):
                         if meta.interp_method is not None:
                             util.interp_masked(data, meta, i, log)
 
+                        # Check if aperture shape has been defined
+                        if (not hasattr(meta, 'aperture_shape') 
+                           or meta.aperture_shape is None):
```

Comment:
For flake8 compliance, you'll need to indent the second line of this if statement by 8 spaces compared to the level of the first line of the if statement

Reply:
Done, sorry. My local linter was being weird about this for some reason.